component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1225,10 +1230,28 @@ private boolean getConnection(HttpURLConnection httpConnection, AuthenticationCo
             } else if (SMSOTPConstants.POST_METHOD.equalsIgnoreCase(httpMethod)) {
                 httpConnection.setRequestMethod(SMSOTPConstants.POST_METHOD);
                 if (StringUtils.isNotEmpty(payload)) {
-                    payload = payload.replaceAll("\\$ctx.num", encodedMobileNo)
-                            .replaceAll("\\$ctx.msg", smsMessage + otpToken)
-                            .replaceAll("\\$ctx.otp", otpToken);
-
+                    String contentType = ((String) headerElementProperties.get(SMSOTPConstants.CONTENT_TYPE)).trim();
+                    /*
+                    If the enable_payload_encoding_for_sms_otp configuration is disabled, mobile number in the
+                    payload will be URL encoded for all the content-types except for application/json content type
+                    preserving the previous implementation to support backward compatibility.
+                    */
+                    if (SMSOTPUtils.isPayloadEncodingForSMSOTPEnabled(context)) {
+                        /*
+                        here only the mobile number and SMS message will be encoded, assuming the rest of the content is
```

Comment:
Start the comment with a capital letter

Reply:
addressed with [31f8d49](https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/136/commits/31f8d49f284a10796db42765fa50021f9a6ebd07) in [https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/136](https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/136)